test/sol6/multipleEpochRewardsClaimer.js

```diff
@@ -0,0 +1,227 @@
+    
+            result = await rewardsClaimer.getUnclaimedEpochs(feeHandler.address, staker);
+            Helper.assertEqual(result.length, zeroBN, "result not empty");
+            await mockKyberDao.setStakerPercentageInPrecision(stakerPercentageInPrecision);
+        });
+
+        it("should return all unclaimed epochs for staker", async() => {
+            let numEpochs = 15;
+            initialEpoch = (await mockKyberDao.epoch()).toNumber();
+            await sendFeesToFeeHandler(mockKyberDao, feeHandler, numEpochs);
+            currentEpoch = (await mockKyberDao.epoch()).toNumber();
+    
+            result = await rewardsClaimer.getUnclaimedEpochs(feeHandler.address, zeroAddress);
+            Helper.assertEqual(result.length, numEpochs, "result.length != numEpochs");
+        });
+
+        it("should return unclaimed epochs if staker claimed for some epochs", async() => {
+            let numEpochs = 5;
+            initialEpoch = (await mockKyberDao.epoch()).toNumber();
+            await sendFeesToFeeHandler(mockKyberDao, feeHandler, numEpochs);
+            currentEpoch = (await mockKyberDao.epoch()).toNumber();
+
+            // claim epochs 2 and 3
+            await feeHandler.claimStakerReward(staker, 2);
+            await feeHandler.claimStakerReward(staker, 3);
+            result = await rewardsClaimer.getUnclaimedEpochs(feeHandler.address, staker);
+            Helper.assertEqualArray(result, [0,1,4], "result != expected unclaimed epochs");
+        });
+    });
+
+    describe("test claimMultipleRewards", async() => {
+        let initialEpoch;
+        let currentEpoch;
+        let numEpochs = 3;
+
+        before("setup dao, feehandler and rewardsClaimer", async() => {
+            // setup mockKyberDao
+            epoch = new BN(0);
+            expiryTimestamp = new BN(5);
+            mockKyberDao = await MockKyberDao.new(
+                rewardInBPS,
+                rebateInBPS,
+                epoch,
+                expiryTimestamp
+            );
+    
+            stakerPercentageInPrecision = precisionUnits.mul(rewardInBPS).div(BPS);
+            await mockKyberDao.setStakerPercentageInPrecision(stakerPercentageInPrecision);
+    
+            // setup feeHandler
+            feeHandler = await FeeHandler.new(daoSetter, proxy.address, kyberNetwork, knc.address, BURN_BLOCK_INTERVAL, daoOperator);
+            await feeHandler.setDaoContract(mockKyberDao.address, {from: daoSetter})
+            await feeHandler.setBurnConfigParams(sanityRate.address, ethWeiToBurn, {from: daoOperator});
+            await feeHandler.getBRR();
+    
+            // setup rewardsClaimer
+            rewardsClaimer = await MultipleEpochRewardsClaimer.new(mockKyberDao.address);
+        });
+
+        beforeEach("send fees for numEpochs", async() => {
+            initialEpoch = (await mockKyberDao.epoch()).toNumber();
+            await sendFeesToFeeHandler(mockKyberDao, feeHandler, numEpochs);
```

Comment:
how do u make sure staker is eligible for all epochs?

Reply:
`await mockKyberDao.setStakerPercentageInPrecision(stakerPercentageInPrecision);`

Reply:
oh
so for mock we don't need to vote
got it.